File: Accenture/10.SecondLargest/code.py
def getSecondLargest(nums):
    if len(nums) < 2: return -1
    largest, sLargest = float('-inf'), float('-inf')
    for num in nums:
        if num > largest:
            sLargest = largest
            largest = num
        elif num > sLargest and num < largest:
            sLargest = num
    return sLargest

def getSecondSmallest(nums):
    if len(nums) < 2: return -1
    smallest, sSmallest= float('inf'), float('inf')
    for num in nums:
        if num < smallest:
            sSmallest = smallest
            smallest = num
        elif num < sSmallest and num != smallest:
            sSmallest = num
    return sSmallest

n = int(input())
nums = list(map(int, input().split()))
secondLargest = getSecondLargest(nums)
secondSmallest = getSecondSmallest(nums)
print("SecondLargest: {}\nSecondSmallest: {}".format(secondLargest, secondSmallest))

File: Accenture/10.SecondLargest/code.cpp
#include<bits/stdc++.h>
using namespace std;
int secondSmallest(int arr[],int n)
{
    if(n<2)
        return -1;
    int small = INT_MAX;
    int secondSmallest = INT_MAX;
    int i;
    for(i = 0; i < n; i++) 
    {
       if(arr[i] < small)
       {
          secondSmallest = small;
          small = arr[i];
       }
       else if(arr[i] < secondSmallest && arr[i] != small)
       {
          secondSmallest = arr[i];
       }
    }
   return secondSmallest;     
}
int secondLargest(int arr[],int n)
{
	if(n<2)
	return -1;
    int large=INT_MIN,second_largest=INT_MIN;
    int i;
    for (i = 0; i < n; i++) 
    {
        if (arr[i] > large) 
        {
            second_largest = large;
            large = arr[i];
        }
 
        else if (arr[i] > second_largest && arr[i] != large) 
        {
            second_largest = arr[i];
        }
    }
    return second_largest;                
}

int main() {
    int arr[]={1,2,4,7,7,5};  
    int n=sizeof(arr)/sizeof(arr[0]);
    int sS=secondSmallest(arr,n);
    int sL=secondLargest(arr,n);
    cout<<"Second smallest is "<<sS<<endl;
    cout<<"Second largest is "<<sL<<endl;
    return 0;
}


